packages/grid/_modules_/grid/models/api/gridColumnApi.ts

```diff
@@ -68,3 +57,22 @@ export interface GridColumnApi {
    */
   setColumnWidth: (field: string, width: number) => void;
 }
+
+export interface GridVisibleColumnApi {
+  /**
+   * Returns the currently visible columns.
+   * @returns {GridStateColDef[]} An array of [[GridStateColDef]].
+   */
+  getVisibleColumns: () => GridStateColDef[];
+  /**
+   * Changes the visibility of the column referred by `field`.
+   * @param {string} field The column to change visibility.
+   * @param {boolean} isVisible Pass `true` to show the column, or `false` to hide it.
+   */
+  setColumnVisibility: (field: string, isVisible: boolean) => void;
+}
+
+/**
+ * The column API interface that is available in the grid [[apiRef]].
+ */
+export type GridColumnApi = GridColumnUpdateApi & GridColumnInformationApi & GridVisibleColumnApi;
```

Comment:
Prefer using an interface, it creates a flat structure, not a union. TypeScript processes it faster
```suggestion
export interface GridColumnApi extends GridColumnUpdateApi, GridColumnInformationApi, GridVisibleColumnApi;
```